### DISCUSSION TITLE
Exporting favourites via Share or Backup can yield different results

### DISCUSSION THREAD

**Discussion by frobijn** [osmandapp/OsmAnd#12807] [Q&A]
### Description
(This is something that is unlikely to occur, I've encountered it while playing around to get [automatic synchronization of waypoints](https://github.com/osmandapp/OsmAnd/issues/2750) to work)

The app uses the file favourites.gpx as source of the Share operation, but queries the internal database when backing up the favourites. The file favourites.gpx file is updated if any of the favourites is added/modified/deleted. However, the file favourites.gpx can be modified by external programs as well. If that is the case, the modified favourites.gpx is shared and not the favourites listed in the app. This is corrected next time one of the favourites is added/modified/deleted, as the favourites.gpx is re-created.

### How to reproduce?
* Replace the existing favourites.gpx file in OsmAnd's data directory by a modified version
* Go to My Places / Favourites
* Share the favourites
* Examine the shared favourites.gpx - it is the modified version.

### Your Environment
OsmAnd Version: OsmAnd+ 4.0.7
Android/iOS version: Android
Device model: not relevant

**Comment by vshcherb** [comment]
Here are some items for your investigation:
1. Database is not used at all https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java#L191 for new devices (there are double if's **internal file** and **database** should exist, so it's only true for pre 3.0 installation old devices
2. There are practically 2 files: **internal** (internal memory) and **external** (external memory) they are constantly getting sync, so it's possible to replace or modify external and get some effects working
3. Sharing / backup could use one of those files, so they could look slightly different but that should not be a problem content-wise.
